sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Linq/DigitalTwinsQueryBuilder.cs

```diff
@@ -0,0 +1,391 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Globalization;
+using System.Text;
+using Azure.Core;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder.Linq
+{
+    /// <summary>
+    /// Azure DigitalTwins Query builder that facilitates writing queries against ADT instances. Redirects to generic version of this class using BasicDigitalTwin as type T.
+    /// </summary>
+    public class DigitalTwinsQueryBuilder : DigitalTwinsQueryBuilder<BasicDigitalTwin>
```

Comment:
We don't like to put more than 1 class in a file. Why are there 2 in here?

Reply:
Good question. 
Ted recommended that we add a class that has the BasicDigitalTwins model as the default generic parameter to the query builder, so if the user doesn't want to use a new type to write the query, they get the BasicDigitalTwin by default. We can't have the same class name and the same file in this namespace, so the two classes live side by side in the same file. One just adds default functionality by providing the generic type.

Reply:
How about DigitalTwinsQueryBuilder.cs and DigitalTwinsQueryBuilderOfT.cs?

https://stackoverflow.com/questions/804036/convention-for-filenames-of-generic-classes

Reply:
Then we have a class that is in a file that is not the same as the class name itself ... which is also a warning we will have to suppress.

Reply:
The `Foo.cs` and `FooOfT.cs` pattern is the one our tooling doesn't complain about.  See https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Response.cs vs. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResponseOfT.cs.  This "one class per file" rule was specifically added per customer feedback about navigating our repos, but I think it's less useful in cases like this where both classes have the same name and they're coupled tightly together.  I just suppress the warning in these cases, but your team can pick whichever matches your style better.